R/pkg/R/DataFrame.R

```diff
@@ -1233,7 +1233,7 @@ setMethod("collect",
                   port = port, blocking = TRUE, open = "wb", timeout = connectionTimeout)
                 output <- tryCatch({
                   doServerAuth(conn, authSecret)
-                  arrowTable <- arrow::read_arrow(readRaw(conn))
+                  arrowTable <- arrow::read_ipc_stream(readRaw(conn))
```

Comment:
This looks good.

Reply:
Quick question. When was this API added? R side currently supports Arrow 0.15.1+.

Reply:
Jenkins has old Arrow, @HyukjinKwon , and this passed Jenkins~

Reply:
Please note that the failure didn't occurred in Jenkins environment. Only GitHub Action and AppVeyor failed so far.

Reply:
Oh.. Jenkins doesn't have arrow? Let me check then.

Reply:
BTW, I believe Apache Spark 3.1 can have Arrow 1.0.0 of course, but we have also a pinning PR if we want.
- https://github.com/apache/spark/pull/29251

Reply:
Ah, Jenkins doesn't have one yet. It's true that people will likely use the latest version of Arrow R as R rather aggressively encourages people to use the latest version. I will just make a quick followup with an if-else.

Reply:
Actually, I think we can simply bump up the minimum Arrow version for SparkR specifically.

Reply:
It exists in arrow_0.17.1.tar. I checked NAMESPACE~

Reply:
Ah, right, this was added since 0.17.0.

Reply:
Oh.. It's not 0.15. Got it.

Reply:
For this part, if we are not bump up our minimum, we need to have `if .. else`. Please make a follow up. Thanks, guys.

Reply:
Maybe increasing minimum Arrow version to 0.17.1?

Reply:
Let me just bump up the minimal version of Arrow in SparkR at https://github.com/apache/spark/pull/29253. Should be fine since such minimal version bump-up is already documented.